common/src/main/java/org/jboss/pnc/common/gerrit/ScmUrlGeneratorProvider.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2022 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.common.gerrit;
+
+import org.jboss.pnc.common.net.GitSCPUrl;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.validation.constraints.NotNull;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+/**
+ * Class to deal with Gerrit/Gitlab information
+ */
+@ApplicationScoped
+public class ScmUrlGeneratorProvider {
```

Comment:
Cover this class with tests please

Reply:
Do you mean to cover `getScmUrlGenerator()`? The `determineScmProvider()` is covered by test cases.

Reply:
sure